### DISCUSSION TITLE
Adding Translations

### DISCUSSION THREAD

**Discussion by paroga** [qudt/qudt-public-repo#580] [Ideas]
Do you accept pull request to add translations (primarily `rdfs:label` for other languages) directly to the vocabulary file?
If not, do you have any other good idea how to share translations? E.g. separate file or additional repository?

**Comment by steveraysteveray** [comment]
A pull request would be an excellent way to add additional languages. I assume you are talking about using the language tag to additional rdfs:label values.

Thanks so much - we need more languages!

**Comment by jhodgesatmb** [comment]
Localizations are generally handled with skos:altLabel, not rdfs:label.
Have we adopted another approach because using additional rdfs:label values
seems confusing.

On Fri, Oct 14, 2022 at 5:41 AM steveraysteveray ***@***.***>
wrote:

> A pull request would be an excellent way to add additional languages. I
> assume you are talking about using the language tag to additional
> rdfs:label values.
>
> Thanks so much - we need more languages!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/qudt/qudt-public-repo/discussions/580#discussioncomment-3878812>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AATQRWN7WCJUMQSSA6QLOL3WDFIIXANCNFSM6AAAAAARFD5LVU>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 
Jack

**Reply by dr-shorthair** [reply]
Multiple `skos:prefLabel` is permitted with different language tags - see https://www.w3.org/TR/skos-reference/#L1629 
I've seen that more often than `altLabel` for language localization.

**Comment by steveraysteveray** [comment]
The advantage of using language tags with rdfs:label is that many tools (including the EDG browser) respond to the language preferences set in your web browser to display your preferred language.

**Comment by jhodgesatmb** [comment]
I would not want the semantic model compromised by this kind of ‘value’, which seems to be driven by special interests. If the semantic community is implementing localization this way, now, then ok. I am used to a class/instance/property having a single label.Jack Hodges, Ph.D.Arbor StudiosOn Oct 14, 2022, at 2:45 PM, steveraysteveray ***@***.***> wrote:﻿
The advantage of using language tags with rdfs:label is that many tools (including the EDG browser) respond to the language preferences set in your web browser to display your preferred language.

**Comment by steveraysteveray** [comment]
At https://www.w3.org/2004/12/q/doc/rdf-labels.html, it states: "Each component should always be assigned a short name. This should be a name suitable for display in a user interface and should be consumer-oriented in nature. A good short name would be "Violence" or "Injury to animals", a bad short name would be "vb" or "vz". RDF provides a mechanism for these short names by using the rdfs:label property. A component can have any number of rdfs:label property values, although it is STRONGLY recommended that they should be distnguished from each other using an xml:lang attribute and that there should be only one label per language."

Look at DBPedia, e.g. [https://dbpedia.org/ontology/Company](https://dbpedia.org/ontology/Company) or [https://dbpedia.org/ontology/Album](https://dbpedia.org/ontology/Album). The use of multiple labels with different language tags is, I believe, well-accepted and widely used.

**Comment by jhodgesatmb** [comment]
So QUDT never uses skos:altLabel? Your quote of the 2004 statement is not
following the usage I have seen in the intervening decades. If QUDT is
going to forsake the use of skos:altLabel then we should document it and
clean up any usage of skos:altLabel in the vocabulary.

On Sat, Oct 15, 2022 at 8:16 AM steveraysteveray ***@***.***>
wrote:

> At https://www.w3.org/2004/12/q/doc/rdf-labels.html, it states: "Each
> component should always be assigned a short name. This should be a name
> suitable for display in a user interface and should be consumer-oriented in
> nature. A good short name would be "Violence" or "Injury to animals", a bad
> short name would be "vb" or "vz". RDF provides a mechanism for these short
> names by using the rdfs:label property. A component can have any number of
> rdfs:label property values, although it is STRONGLY recommended that they
> should be distnguished from each other using an xml:lang attribute and that
> there should be only one label per language."
>
> Look at DBPedia, e.g. https://dbpedia.org/ontology/Company or
> https://dbpedia.org/ontology/Album. The use of multiple labels with
> different language tags is, I believe, well-accepted and widely used.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/qudt/qudt-public-repo/discussions/580#discussioncomment-3885747>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AATQRWIQ6DEK7A3FHLJAQXLWDLDGJANCNFSM6AAAAAARFD5LVU>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Jack

**Comment by jhodgesatmb** [comment]
The policy seems confusing in a couple of places. Using:

*SELECT* *

*WHERE* {

    ?cls rdfs:subClassOf* qudt:Concept .

    *OPTIONAL* {?cls skos:altLabel ?caltl} .

    ?inst a ?cls .

    ?inst skos:altLabel ?ialtl .

}

I find unit:MilliM, unit:MegaYR, unit:MilliN where the policy is quite
variable. I recommend that we publish the policy you are putting forth and
resolve these discrepancies. Fortunately there do not appear to be many. We
should also update our submission guidelines to reflect that localizations
should be implemented this way.

Jack

On Sat, Oct 15, 2022 at 9:20 AM Jack Hodges ***@***.***> wrote:

> So QUDT never uses skos:altLabel? Your quote of the 2004 statement is not
> following the usage I have seen in the intervening decades. If QUDT is
> going to forsake the use of skos:altLabel then we should document it and
> clean up any usage of skos:altLabel in the vocabulary.
>
> On Sat, Oct 15, 2022 at 8:16 AM steveraysteveray ***@***.***>
> wrote:
>
>> At https://www.w3.org/2004/12/q/doc/rdf-labels.html, it states: "Each
>> component should always be assigned a short name. This should be a name
>> suitable for display in a user interface and should be consumer-oriented in
>> nature. A good short name would be "Violence" or "Injury to animals", a bad
>> short name would be "vb" or "vz". RDF provides a mechanism for these short
>> names by using the rdfs:label property. A component can have any number of
>> rdfs:label property values, although it is STRONGLY recommended that they
>> should be distnguished from each other using an xml:lang attribute and that
>> there should be only one label per language."
>>
>> Look at DBPedia, e.g. https://dbpedia.org/ontology/Company or
>> https://dbpedia.org/ontology/Album. The use of multiple labels with
>> different language tags is, I believe, well-accepted and widely used.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/qudt/qudt-public-repo/discussions/580#discussioncomment-3885747>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AATQRWIQ6DEK7A3FHLJAQXLWDLDGJANCNFSM6AAAAAARFD5LVU>
>> .
>> You are receiving this because you commented.Message ID:
>> ***@***.***>
>>
>
>
> --
> Jack
>


-- 
Jack

**Comment by steveraysteveray** [comment]
Done.
